lib/relation-definition.js

```diff
@@ -1369,6 +1411,20 @@ BelongsTo.prototype.related = function (refresh, params) {
   }
 };
 
+
+/**
+ * Define a Promise-based method for the belongsTo relation itself
+ * - order.customer.get(refresh): Load the target model instance asynchronously
+ *
+ * @param refresh
+ * @returns {Promise}
+ */
+BelongsTo.prototype.get = function (refresh) {
+  var cb = utils.createPromiseCallback();
+  return this.related(refresh, cb);
+}
```

Comment:
It is almost the same, except it does not support zero-arg sync version returning a cached copy.

At the moment, `order.user()` returns a cached copy of the related user, thus it's not possible to return a promise instead. See https://github.com/strongloop/loopback/issues/418#issuecomment-74442529 and https://github.com/strongloop/loopback/issues/418#issuecomment-74691763

I find `order.user(false).then...` and `order.user(true).then...` difficult to read, what the heck is true/false supposed to mean without reading the docs? Therefore I proposed to add a new clean API that does not suffer from this problem.